### PR TITLE
Script to generate failures histogram

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #118008

Generates something that looks like
https://gist.github.com/zou3519/43aa8ef28a327bd68cfbac83d84c0999

### CODE REVIEW

File: scripts/compile_tests/failures_histogram.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+def skip_reason(testcase):
+    for child in testcase.iter():
+        if child.tag != "skipped":
+            continue
+        return child.attrib["message"]
+    raise AssertionError("no message?")
+
+
+IGNORED_REASONS = {
```
Comment (oulgen) [inline scripts/compile_tests/failures_histogram.py:24]: Should we mention why we skip some of these? Like e.g. why is "Skipped" or "Skipped!" skipped?

File: scripts/compile_tests/failures_histogram.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    "NP_VER: fail on CI",
+    "Excluded from CUDA tests",
+    "CUDA not found",
+    "Only runs on cuda",
+    "NumPy not found",
+    "test is slow; run with PYTORCH_TEST_WITH_SLOW to enable test",
+    "Skipped",
+    "Skipped!",
+    "",
+    "test doesn't currently work with dynamo",
```
Comment (oulgen) [inline scripts/compile_tests/failures_histogram.py:36]: this seems like a good reason to fix?

**Review by yanboliang — APPROVED** [review]

File: scripts/compile_tests/failures_histogram.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        skipped_dict[reason].append(s)
+    result = []
+    for s, v in skipped_dict.items():
+        result.append((len(v), s, v))
+    result.sort(reverse=True)
+    return result
+
+
+def repro(testcase):
+    return f"PYTORCH_TEST_WITH_DYNAMO=1 pytest {testcase.attrib['file']} -v -k {testcase.attrib['name']}"
```
Comment (oulgen) [inline scripts/compile_tests/failures_histogram.py:73]: for future, might be good to give file path relative to project root

**Review by oulgen — APPROVED** [review]

File: scripts/compile_tests/failures_histogram.py (hunk 2 above)
Comment (zou3519) [reply scripts/compile_tests/failures_histogram.py:36]: I should probably just delete most things from this list.

**Pull request closed without merging**